test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,415 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale
+{
+    [Trait(TestTraits.CategoryTraitName, TestTraits.ScaleMonitoring)]
+    public class ScaleHostEndToEndTests
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+        private const string Function3Name = "Function3";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabled)
+        {
+            var triggerMetadata = new List<TriggerMetadata>()
+            {
+                new TriggerMetadata(Function1Name, new JObject { { "type", "testExtensionATrigger" } }),
+                new TriggerMetadata(Function2Name, new JObject { { "type", "testExtensionATrigger" } }),
+                new TriggerMetadata(Function3Name, new JObject { { "type", "testExtensionBTrigger" } }, new Dictionary<string, object> { { "foo", "bar" } }),
+            };
+
+            JArray functionMetadata = new JArray
+            {
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function1Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function2Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionBTrigger" },
+                    { "functionName", Function3Name }
+                }
+            };
+
+            string hostJson =
+            @"{
+              ""extensions"": {
+                ""testExtensionA"" : {
+                  ""foo"": 16
+                },
+                ""testExtensionB"" : {
+                  ""bar"": 8
+                }
+              }
+            }";
+
+            string hostId = "test-host";
+            var loggerProvider = new TestLoggerProvider();
+
+            IHostBuilder hostBuilder = new HostBuilder();
+            hostBuilder.ConfigureLogging(configure =>
+            {
+                configure.SetMinimumLevel(LogLevel.Debug);
+                configure.AddProvider(loggerProvider);
+            });
+            hostBuilder.ConfigureAppConfiguration((hostBuilderContext, config) =>
+            {
+                // Adding host.json here
+                config.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(hostJson)));
+
+                // Adding app settings
+                config.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { "app_setting1", "value1" },
+                    { "app_setting2", "value2"},
+                    { "FeatureManagement:Microsoft.Azure.WebJobs.Host.EndToEndTests", "true" }
```

Comment:
[Mathew]: Here's how a feature flag can be added. We can also control the root "FeatureManagement" name - we could have it be "WebJobsFeatures" or something.

[Alexey]: The root name should be "HostingConfig" as function host already use it.

Reply:
It must be the same as in function host. In function host we are using "[FunctionsHostingConfig](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/ScriptConstants.cs#L224)". But it can be changes to whatever name we like.